core/src/main/scala/org/apache/spark/internal/io/HadoopMapReduceCommitProtocol.scala

```diff
@@ -57,6 +57,11 @@ class HadoopMapReduceCommitProtocol(jobId: String, path: String)
    */
   private def absPathStagingDir: Path = new Path(path, "_temporary-" + jobId)
 
+  /**
+   * Checks whether there are files to be committed to an absolute output location.
+   */
+  private def hasAbsPathFiles: Boolean = addedAbsPathFiles != null && addedAbsPathFiles.nonEmpty
```

Comment:
When addedAbsPathFiles is null and when it is not is slightly confusing.
Can we move to using an Option[Map[String, String]] instead ?

Reply:
Good catch, thank you!
According to the `FileCommitProtocol`, `addedAbsPathFiles` is always null on driver, so  we will not be able to commit or remove these files.

Replaced it with

     private def hasAbsPathFiles: Boolean = path != null